### ISSUE TITLE
ExperimentalKeepIndentInBranch should fire when first branch was all on one line

### ISSUE THREAD

**Issue by Smaug123** [fsprojects/fantomas#2973]
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/main?data=N4KABGBEDGD2AmBTSAuKBbBBXANosACgE6wDmRAhumALwA6AdhBHgC5joUCWTA%2BrY2ZCwXAGZhRFHAGd8rABaImARjCIZiQcIgAHIj1aimdSPK4BCE1u0AGSABpwUWa1Y9S01GADaTiKG1IABJWAE8dZDRIAwc-ZmCANyksSJ84oRtHbQhIAEkGJAZWAGUuAC9kLOyAFnSwAF04gF8q-zrgsIivaKLYwKCknBSvX2ywZVbhSABZCgAPABkeRAWlUgU%2B7OUAJhs6xqEWuICpkPDUyCV4AHlRJYZEYrC8TaFE5NTR7O3Jt4BRAq3e6VOo5HCiSD7ZqTE5vM5dKIAI1gsBwr3iAw%2BI1BYAAzL94vlZERWAAxHhSAByiAA7jhluihKwiCkoYcYe14RdkajGVBMUNPjjqgScsUdBRoIgAEKIUSwIiIAgUSjoRCsRBEPkQZms7QHZhHISwjGdbkotEE96C7FjACsoqg4slMrlCpWsBpmugFFk%2BQSsB9blgDG1YF1iDZho5-TN3R5ls5g2GaC%2B2gAbI7IM6pbL5YqAKo6CJEH1%2BhgBoNcENhyQaKMQI3ME05Lnxi1860ptJjADsWZzrvziAAwjhfdIRyHpLroKwFbWpLIG2Am21Y%2Bd27yrQLu2nhAAOAcS3NuxXTRDoRGaxf1-XQ46cuNIjs75NCsYATmPLrz7qnOA1o6dbLve7KPhuCJQAmna7h%2BWyZDi2YnogACCogakQU7oJwYYRiua5gC2-LPtBr5JliqY4soExIYOf6KsUl5cHAgGhsBS6RmB0YQacpGQDBb6UT2Ww-HRKHoZhTHoCxqJATi%2BHcY2MZ8ZuL7bhRNpUWMyj4uJLqoSQWAFAAIuoXAyZheEslxwgGspvFwvxMRCVpInaMoIpIbMcy5KIAAqigMMU8gKqwADqXDwBsjp7Epq4qU5alQC5ml7tRDrefMfmBUofwaCFYWRdF8hhumcV2Q%2BxpPslPSsLB762lsmZZb5DCiFwczXCWFDzkQfxzHoiDSNI1bsTiB4VUI9kJY5pq1al-SNdpWz9q1ABKiBwEQ8DFTFwpTcwM2EcRHQLb0rnpTpR4bVtCrwJSWBXpqty5Bq6CeI6ygEYl81QZA6C4G49IPKSCqcK4mr%2BZul3wR535IZt23wNMQNcCDiBg0QENWVm0DyCqkqYbwNJRRsP1za2zkXWlcPCLsWY%2BYZlChNcRBLDOe2lY6k0U9VkEXItpzLe59O0WMMzzMzFCs%2BzXAzo9z1EK972fdRfPNjV-2AzgwPLFjONQzDtNNdo2xiRL0uyxzrCo7r6P6%2BDvW40h%2BOE3Omok2TpUa%2Buqn-ULcIi-uQjbHpEs%2BQAah80o8PA7hc2GvPxSdWuCzTS3CSHzDbF5EfzKSxlzmNscFAn3thtUh0QMdv1U%2Bd9Ww6b9OZfncwmbArAAOLqgNQ0jWNic89XDRVZrAtbommdudnEDbC1EvUnSyyyqwXpKNDERmR1DBcMGDCoQUF5K2rYyKZV4H8-75oadPV3fGtlv0qQDCFwwxchqUL%2B9Vgir%2BbA%2BRCisF6mNW8oEL48Svklf6gkTYrTNjdJ%2BmEGAgISB6BgpBj7XiINvHge8xqn2yCBWy00x5%2B2gTfKewss44m2AjCOaNgQLCoIieAFAxywFGhgpeGMwEkKOmQoiadJ4NRoWMXEiEJZ901BZJQwCcAAGlECIB0IAuR%2BRpSUHftzBSNlfZCInupKhQcxHZFxOLbIkBpTjgYAAa2BNIQysBjIPWGhqFGaMMZ90VAPac1k9QQIclAv6lDREzxxLiC2ljpQqgYogEy8toD6Bkig9xBZd4hjMtAcclB958P0adNsUQdZ6weJoyUtj1RPFCC8Ju8DhC4nDpYu2pSZSUGgJUkozwQQSySVQCI8BIQpzriRBuYT77aFxHnSxSiVE%2BUVtg241iKB2IcWGZQNgR610pqMmB5E750yELiVulieErzPIgFpDsHjYR0FgYB%2B9vHDVGvJM%2Bejhk7LOns2%2B1DwniIXpY6RyS5FSHyjJaQOixjEIIk4eoVRojSFJKNLw0KmhAA)

#### Code

```fsharp
module Program =
    let main _ =
        if false then 1 else
        printfn "hi!"
        0
```
            

#### Result

```fsharp
module Program =
    let main _ =
        if false then
            1
        else
            printfn "hi!"
            0

```

#### Desired result

```fsharp
module Program =
    let main _ =
        if false then
            1
        else
        printfn "hi!"
        0

```
            
#### Problem description

This is a completely new request which we've never asked you for before, not a bug. (In general the `ExperimentalKeepIndentInBranch` feature works great nowadays, and I now instinctively expect it to do the right thing!)

If ExperimentalKeepIndentInBranch, it would be great if we could retain the indentation when there's a one-line "deal with some particular case" that we've reformatted onto multiple lines. I think this new pattern should *only* fire if the format is like this:

```fsharp
if blah then baz else
more stuff at the same level of indentation as the initial `if`
```

That is, the following reformatting *shouldn't* take place:

```fsharp
if blah then baz else foo

//

if blah then
    baz
else
foo // note the lack of indent, which here would be incorrect
```

Rationale: if I've "clearly been trying" to keep the main body of my `else` block unindented, by explicitly putting it at the same level of indentation as the `if` keyword, then I think we shouldn't break that attempt merely because we reformatted the body of the `if` branch.

The actual code in the wild which triggered me to write this report is:

```fsharp
async {
    if not (proc.Start ()) then return Error "failed to start" else
    use stdout = proc.StandardOutput
    let! ct = Async.CancellationToken
    ...
}
```

#### Extra information

- [ ] The formatted result breaks my code.
- [ ] The formatted result gives compiler warnings.
- [ ] I or my company would be willing to help fix this.
- [ ] I would like a release if this problem is solved.

#### Options

Fantomas main branch at 2023-10-30T10:39:28Z - 3f293e712175343c2bcab1dfabe35cbb52fe8839

```fsharp
    { config with
                ExperimentalKeepIndentInBranch = true }
```

<sub>Did you know that you can ignore files when formatting by using a [.fantomasignore file](https://fsprojects.github.io/fantomas/docs/end-users/IgnoreFiles.html)?</sub>
<sub>PS: It's unlikely that someone else will solve your specific issue, as it's something that you have a personal stake in.</sub>

**Comment by nojaf** [comment]
Hi Patrick,

I've encountered this myself as well. I see your point.
The additional trigger should be the `elseBranch` matching the indentation of the `if` keyword right?
I'll take a look at this.

**Comment by Smaug123** [comment]
Yep, exactly.

**Comment by nojaf** [comment]
I released a new preview version with this: https://www.nuget.org/packages/fantomas/6.3.0-alpha-001

**Comment by josh-degraw** [comment]
Is this related to #1361?

**Comment by nojaf** [comment]
Yes, see [fsharp_experimental_keep_indent_in_branch](https://fsprojects.github.io/fantomas/docs/end-users/Configuration.html#fsharp_experimental_keep_indent_in_branch)

**Comment by nojaf** [comment]
This works now with fsharp_experimental_keep_indent_in_branch
